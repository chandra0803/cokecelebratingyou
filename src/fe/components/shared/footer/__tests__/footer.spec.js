/*global jest*/

jest.mock( 'react-redux' );
import React from 'react';
import Footer from '../footer.js';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
const defaultProps = {
    content: [
    {
        key: 'COPYRIGHT_TEXT',
        code: 'system.general',
        content: 'copyright'
    },
    {
        key: 'CONTACT_US',
        code: 'system.general',
        content: 'contact'
    },
    {
        key: 'T&C',
        code: 'system.general',
        content: 'T&C'
    },
    {
        key: 'PRIVACY_POLICY',
        code: 'system.general',
        content: 'privacy'
    }
],
showTerms: true
};
test( 'Footer renders properly', () => {
    const props = {
        ...defaultProps
    };
    const component = renderer.create(
        <div>
            <Footer { ...props }/>
        </div>
    );

    const tree = component.toJSON();
    expect( tree ).toMatchSnapshot();
} );
it ( 'toggles help modal when help link is clicked.', () => {
    const props = {
        ...defaultProps,
        toggleModalDisplay: jest.fn()
    };

    const tree = mount( <Footer { ...props } /> );

    const link = tree.find( '#privacyPolicyFooterLink' );

    link.simulate( 'click' );

    expect ( props.toggleModalDisplay.mock.calls.length ).toBe( 1 );
} );
