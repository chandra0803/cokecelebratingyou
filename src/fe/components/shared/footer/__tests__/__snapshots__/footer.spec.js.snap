// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders properly 1`] = `
<div>
  <footer
    className="site-header"
    id="footer"
  >
    <div
      className="container"
    >
      <div
        id="globalFooterView"
      >
        <ul
          className="nav"
        >
          <li>
            <a
              id="privacyPolicyFooterLink"
              onClick={[Function]}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "system.general.privacy_policy",
                  }
                }
              />
            </a>
          </li>
          <li>
            <a
              href="termsAndConditionsView.do?method=review"
              id="termsFooterLink"
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "system.general.t&c",
                  }
                }
              />
            </a>
          </li>
          <li
            className="copyright"
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "system.general.copyright_text",
                }
              }
            />
          </li>
        </ul>
      </div>
    </div>
  </footer>
</div>
`;
